src/components/dataset/query/sidebar/QuerySidebarPanel.jsx

```diff
@@ -28,33 +28,51 @@ export class QuerySidebarPanel extends React.PureComponent {
     filterData: PropTypes.object,
   };
 
-  clearFilter = filter => {
+  clearFilter = (filter, datum) => {
     const { dispatch, filterData } = this.props;
-    const clonedData = _.clone(filterData);
-    delete clonedData[filter];
+    const clonedData = _.cloneDeep(filterData);
+    const clonedFilter = clonedData[filter];
+    if (Array.isArray(clonedFilter)) {
+      clonedFilter.splice(clonedFilter.indexOf(datum), 1);
+    }
+    if (_.isPlainObject(clonedFilter)) {
+      delete clonedFilter[datum];
+    }
     dispatch(applyFilters(clonedData));
   };
-
   render() {
     const { classes, filterData } = this.props;
     const listFilters = _.keys(filterData).map(filter => {
       const data = _.get(filterData, filter);
       let dataString = data;
       if (Array.isArray(data)) {
-        dataString = _.join(data, ', ');
+        dataString = data.map(datum => {
+          return (
+            <ListItemText>
+              {datum}
+              <Button onClick={() => this.clearFilter(filter, datum)}>
+                <HighlightOff />
+              </Button>
+            </ListItemText>);
```

Comment:
is there a way we can reuse this since it's returning the same case for the array and plain object cases? maybe we can set dataString based on whether or not it's an array and then we can just `map` over it once?

Reply:
Yeah I think we can assign something like `const list = Array.isArray(data) ? data : _.keys(data);` and then map over that if data is not a string? Would DRY this up nicely.